server/app/views/admin/programs/ProgramBlockAllAspectsView.java

```diff
@@ -199,38 +224,34 @@ private DivTag addFormEndpoints(InputTag csrfTag, long programId, long blockId)
   private DivTag blockOrderPanel(Request request, ProgramDefinition program, long focusedBlockId) {
     DivTag ret = div().withClasses("shadow-lg", "pt-6", "w-2/12", "border-r", "border-gray-200");
     ret.with(
-        renderBlockList(
-            request, program, program.getNonRepeatedBlockDefinitions(), focusedBlockId, 0));
+        blockList(request, program,
+          program.getNonRepeatedBlockDefinitions(), focusedBlockId, 0));
 
-    ret.with(
+    if (viewAllowsEditingProgram()) {
+      ret.with(
         ViewUtils.makeSvgTextButton("Add screen", Icons.ADD)
-            .withClasses(AdminStyles.SECONDARY_BUTTON_STYLES, "m-4")
-            .withType("submit")
-            .withId("add-block-button")
-            .withForm(CREATE_BLOCK_FORM_ID));
+          .withClasses(AdminStyles.SECONDARY_BUTTON_STYLES, "m-4")
+          .withType("submit")
+          .withId("add-block-button")
+          .withForm(CREATE_BLOCK_FORM_ID));
+    }
     return ret;
   }
 
-  private DivTag renderBlockList(
+  private DivTag blockList(
```

Comment:
method names should typically start with a verb like render which was removed.

Reply:
I generally agree. The reason I changed this is because the entire rest of the class sticks to the pattern of not using a verb. There are some other bits which aren't great, for example, the signatures are much more complicated than they need to be.

In the initial change that used inheritance I included some cleanups that simplify the code and intermitendly also started renaming the methods, but it increased the size of the change quite a bit.Considering how long this change has been dangling, I decided not to include the cleanups here and track them instead.

Basically the idea was: step 1, consistency in the file, step 2 rename all methods to stick to conventions.

Reply:
Let's stick-with/go with the verb route as it's is the correct one and this was pre-existing too; IE we shouldn't correct inconsistency towards the incorrect route.

Generally we normalize to inconsistent style when we've affirmatively decided it's better, but that's not the case here.

Reply:
done